common/base/include/claragenomics/utils/device_preallocated_allocator.cuh

```diff
@@ -221,8 +227,11 @@ private:
                                                    });
         assert(block_to_be_freed_iter != std::end(used_blocks_));
 
-        // ** wait for all work on associated_stream to finish before freeing up this memory block
-        CGA_CU_ABORT_ON_ERR(cudaStreamSynchronize(block_to_be_freed_iter->associated_stream));
+        // ** wait for all work on associated_streams to finish before freeing up this memory block
```

Comment:
How about adding a comment like "Warning: Please do not rely on this cudaStreamSynchronize in the caller. The allocator implementation may guarantee the lifetime of the memory block without this implicit synchronization in the future."?

Reply:
done